spec/javascripts/packages/device/index-spec.js

```diff
@@ -1,26 +1,93 @@
-import { isLikelyMobile, hasMediaAccess, isCameraCapableMobile } from '@18f/identity-device';
+import {
+  isLikelyMobile,
+  hasMediaAccess,
+  isCameraCapableMobile,
+  isIPad,
+} from '@18f/identity-device';
+
+describe('isIPad', () => {
+  let originalUserAgent;
+  beforeEach(() => {
+    originalUserAgent = navigator.userAgent;
+    navigator.maxTouchPoints = 0;
+    Object.defineProperty(navigator, 'userAgent', {
+      configurable: true,
+      writable: true,
+    });
+    Object.defineProperty(navigator, 'maxTouchPoints', {
+      writable: true,
+    });
+  });
+
+  afterEach(() => {
+    navigator.userAgent = originalUserAgent;
+  });
+
+  it('returns true if ipad is in the user agent string (old format)', () => {
+    navigator.userAgent =
+      'Mozilla/5.0(iPad; U; CPU iPhone OS 3_2 like Mac OS X; en-us) AppleWebKit/531.21.10 (KHTML, like Gecko) Version/4.0.4 Mobile/7B314 Safari/531.21.10';
+
+    expect(isIPad()).to.be.true();
+  });
+
+  it('returns false if the user agent is Macintosh but with 0 maxTouchPoints', () => {
+    navigator.userAgent =
+      'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36';
+
+    expect(isIPad()).to.be.false();
+  });
+
+  it('returns true if the user agent is Macintosh but with 5 maxTouchPoints', () => {
+    navigator.userAgent =
+      'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36';
+    navigator.maxTouchPoints = 5;
```

Comment:
Related to my prior comment, I don't see that we're restoring the original value after this test is run.

Reply:
I've gone ahead and updated the whole file to use the test helper (https://github.com/18F/identity-idp/pull/6865/commits/f24deca94a533c4725cee55e7a96cae175f69b02). LMK what you think

Reply:
After playing around with the test helper for a couple of hours this afternoon, it seems like it won't play nice with `navigator` and the need to set write permissions on the object (and then turn them off).

Reply:
I think there were two issues with the previous commit:

1. You were manipulating some global properties outside test lifecycle (particularly [this one](https://github.com/18F/identity-idp/blob/7dab8e2c9277a7e52c8c211898876f18f824b72c/spec/javascripts/packages/device/index-spec.js#L115-L118))
2. Redefining the properties required using [the `configurable` option](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#configurable)

I was able to get it to pass after those were fixed up.

```js
import {
  isLikelyMobile,
  hasMediaAccess,
  isCameraCapableMobile,
  isIPad,
} from '@18f/identity-device';
import { useDefineProperty } from '@18f/identity-test-helpers';

describe('isIPad', () => {
  const defineProperty = useDefineProperty();

  context('iPad is in the user agent string (old format)', () => {
    beforeEach(() => {
      defineProperty(navigator, 'userAgent', {
        configurable: true,
        value:
          'Mozilla/5.0(iPad; U; CPU iPhone OS 3_2 like Mac OS X; en-us) AppleWebKit/531.21.10 (KHTML, like Gecko) Version/4.0.4 Mobile/7B314 Safari/531.21.10',
      });
    });

    it('returns true', () => {
      expect(isIPad()).to.be.true();
    });
  });

  context('The user agent is Macintosh but with 0 maxTouchPoints', () => {
    beforeEach(() => {
      defineProperty(navigator, 'userAgent', {
        configurable: true,
        value:
          'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36',
      });
    });

    it('returns false', () => {
      expect(isIPad()).to.be.false();
    });
  });

  context('The user agent is Macintosh but with 5 maxTouchPoints', () => {
    beforeEach(() => {
      defineProperty(navigator, 'userAgent', {
        configurable: true,
        value:
          'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36',
      });
      defineProperty(navigator, 'maxTouchPoints', { configurable: true, value: 5 });
    });

    it('returns true', () => {
      expect(isIPad()).to.be.true();
    });
  });

  context('Non-Apple userAgent, even with 5 maxTouchPoints', () => {
    beforeEach(() => {
      defineProperty(navigator, 'userAgent', {
        configurable: true,
        value:
          'Mozilla/5.0 (Linux; Android 13) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.5195.58 Mobile Safari/537.36',
      });
      defineProperty(navigator, 'maxTouchPoints', { configurable: true, value: 5 });
    });
    it('returns false', () => {
      expect(isIPad()).to.be.false();
    });
  });
});

describe('isLikelyMobile', () => {
  const defineProperty = useDefineProperty();

  context('Is not mobile and has no touchpoints', () => {
    beforeEach(() => {
      defineProperty(navigator, 'userAgent', {
        configurable: true,
        value:
          'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36',
        writable: true,
      });
      defineProperty(navigator, 'maxTouchPoints', { configurable: true, value: 0 });
    });

    it('returns false', () => {
      expect(isLikelyMobile()).to.be.false();
    });
  });

  context('There is an Apple user agent and 5 maxTouchPoints', () => {
    beforeEach(() => {
      defineProperty(navigator, 'userAgent', {
        configurable: true,
        value:
          'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36',
      });
      defineProperty(navigator, 'maxTouchPoints', { configurable: true, value: 5 });
    });

    it('returns true', () => {
      expect(isLikelyMobile()).to.be.true();
    });
  });

  context('There is an explicit iPhone user agent', () => {
    beforeEach(() => {
      defineProperty(navigator, 'userAgent', {
        configurable: true,
        value:
          'Mozilla/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148',
      });
    });

    it('returns true', () => {
      expect(isLikelyMobile()).to.be.true();
    });
  });
});

describe('hasMediaAccess', () => {
  const defineProperty = useDefineProperty();

  context('No media device API access', () => {
    beforeEach(() => {
      defineProperty(navigator, 'mediaDevices', { configurable: true, value: undefined });
    });

    it('returns false', () => {
      expect(hasMediaAccess()).to.be.false();
    });
  });

  it('Has media device API access', () => {
    beforeEach(() => {
      defineProperty(navigator, 'mediaDevices', { configurable: true, value: {} });
    });

    it('returns true', () => {
      expect(hasMediaAccess()).to.be.true();
    });
  });
});

describe('isCameraCapableMobile', () => {
  const defineProperty = useDefineProperty();

  context('Is not mobile', () => {
    beforeEach(() => {
      defineProperty(navigator, 'userAgent', {
        configurable: true,
        value:
          'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36',
      });
      defineProperty(navigator, 'mediaDevices', { configurable: true, value: {} });
    });

    it('returns false', () => {
      expect(isCameraCapableMobile()).to.be.false();
    });
  });

  context('No media device API access', () => {
    beforeEach(() => {
      defineProperty(navigator, 'userAgent', {
        configurable: true,
        value:
          'Mozilla/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148',
      });
      defineProperty(navigator, 'mediaDevices', { configurable: true, value: undefined });
    });

    it('returns false', () => {
      expect(isCameraCapableMobile()).to.be.false();
    });
  });

  context('Is likely mobile and media device API access', () => {
    beforeEach(() => {
      defineProperty(navigator, 'userAgent', {
        configurable: true,
        value:
          'Mozilla/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148',
      });
      defineProperty(navigator, 'mediaDevices', {});
    });

    it('returns true', () => {
      expect(isCameraCapableMobile()).to.be.true();
    });
  });
});
```